tests/conftest.py

```diff
@@ -43,3 +48,45 @@ def random_dataset(draw):
     dataset = torch.utils.data.TensorDataset(dataset_x, dataset_y)  # type: ignore
 
     return dataset, num_classes, way
+
+
+def use_basic_embedding_model(random_dataset):
+    dataset, num_classes, _ = random_dataset
+    X, _ = dataset.tensors
+    embedding_model = BasicEmbeddingModel(X.shape[1], num_classes)
+    return dataset, num_classes, X, embedding_model
+
+def use_save_load_model_tests(model, X, tmp_filename:str="tmp_eq.pt"):
```

Comment:
Make sure the tmp file names are different to avoid race conditions if tests are run in parrallel